backend/apps/volontulo/views/api.py

```diff
@@ -65,6 +69,62 @@ def login_view(request):
     )
 
 
+@api_view(['POST'])
+@authentication_classes((CsrfExemptSessionAuthentication,))
+@permission_classes((AllowAny,))
+def register_view(request):
+    """REST API register view."""
+    if not request.user.is_authenticated():
+        email = request.data.get('email')
+        password = request.data.get('password')
+
+        try:
+            user = User.objects.create_user(
+                username=email,
+                email=email,
+                password=password,
+                is_active=False
+            )
+
+            user.save()
+            profile = UserProfile(user=user)
+            ctx = {'token': profile.uuid, 'angular_root':
+                   settings.ANGULAR_ROOT}
+            profile.save()
+            send_mail(request, 'registration', [user.email], context=ctx)
+        except IntegrityError:
```

Comment:
If you catch an exception, catch precisely the line(s) you expect it to occur. I.e. it won't probably occur on the `create_user` or `send_mail`, but on `user.save()` instead. This way we might accidently omit some errors.

Reply:
+1